### DISCUSSION TITLE
How to know if the Ctrl key has been pressed?

### DISCUSSION THREAD

**Discussion by pychans** [flet-dev/flet#1699] [General]
I found that only by combining keys can I know if Ctrl has been pressed (such as Ctrl+C), and is there currently a way to know if individual Ctrl, Shift, Windows, and Alt keys have been pressed? I want to combine the Ctrl key with the mouse.
```python
import flet
from flet import Page, Text, KeyboardEvent


def main(page: Page):
    def on_keyboard(e: KeyboardEvent):
        page.add(
            Text(
                f'Key: {e.key}, Shift: {e.shift}, Control: {e.ctrl}, Alt: {e.alt}, Meta: {e.meta}'
            )
        )

    page.on_keyboard_event = on_keyboard
    page.add(
        Text('Press any key with a combination of CTRL, ALT, SHIFT, META keys and then press Enter.')
    )


flet.app(target=main)
```

**Comment by FeodorFitsner** [comment]
Could you describe your use case please? If it's GestureDetector then we may need to add pressed keys array to gesture events, like here: https://stackoverflow.com/a/71853874/1435891

**Reply by pychans** [reply]
Sorry, maybe I'm not clear, I mean when I press the Ctrl key on the keyboard (only press this one), how to get this state in the program?

**Reply by FeodorFitsner** [reply]
I could imagine some new method like `page.get_pressed_keys()`, but it could be a performance issue if you going to constantly poll it. You are saying "combine the Ctrl key with the mouse" - could you please elaborate on your use case in more details?

**Reply by pychans** [reply]
For example, hold down the Ctrl key and scroll the mouse wheel up to zoom in on the image.

**Reply by FeodorFitsner** [reply]
Got it. That what I meant by offering pressed keys information could be a part of GestureDetector events.

**Reply by pychans** [reply]
Sorry, I read the document and there is no information on this(hold down the Ctrl key and scroll the mouse wheel up to zoom in on the image). The main issue is that KeyboardEvent can't get the separate ctrl, shift, alt, and meta keys.Can we look into the implementation of this?

**Reply by FeodorFitsner** [reply]
Correct, because this feature is not yet implemented.

**Reply by ndonkoHenri** [reply]
I think an issue should be opened for future implementation.

**Comment by marc-svg** [comment]
Hi,
For my desktop app the user should be able to select a control with the mouse but only when Ctrl key is down. Is this possible ? 

Thanks

**Comment by marc-svg** [comment]
Hi,
I am still having that need, need to be able to know if Ctrl key is down when mouse button is down.
Today Ctrl is triggered only if another key is down.
Perhaps is it possible to add keyboard package from pypi ?

Best regards

**Comment by marc-svg** [comment]
Ok keyboard package from Pypi does the job for me.
Its a package I can add to my desktop app.

Sorry for the inconvenience.

Best regards
